File: ChainOfResponsibility/Java/InfoLogger.java
public class InfoLogger extends Logger {
    InfoLogger(Logger nextLogProcessor) {
        super(nextLogProcessor);
    }

    @Override
    public void log(int logLevel, String message) {
        if (logLevel == INFO) {
            System.out.println("INFO: " + message);
        }
        if (nextLogProcessor != null) {
            nextLogProcessor.log(logLevel, message);
        }
    }
}

File: ChainOfResponsibility/C++/InfoLogger.cpp

#include <iostream>
#include "InfoLogger.hpp"


InfoLogger::InfoLogger(Logger* nextLoggerProcessor) : Logger(nextLoggerProcessor)
{
}

void InfoLogger::logMessage(int logLevel, std::string message)
{
    if (logLevel == Logger::INFO)
    {
        std::cout << "INFO: " << message << std::endl;
    }
    else
    {
        Logger::logMessage(logLevel, message);
    }
}
